x-pack/plugins/monitoring/server/alerts/base_rule.ts

```diff
@@ -194,11 +194,6 @@ export class BaseRule {
         const filteredAlertInstance = this.filterAlertInstance(alertInstance, filters);
         if (filteredAlertInstance) {
           accum[instanceId] = filteredAlertInstance as RawAlertInstance;
-          if (filteredAlertInstance.state) {
-            accum[instanceId].state = {
-              alertStates: (filteredAlertInstance.state as AlertInstanceState).alertStates,
-            };
-          }
```

Comment:
`RawAlertInstance` is now a ReadOnly type, since this code was doing a mutation, TypeScript was complaining. This code felt unnecessary so I decided to remove it instead of fixing it..

Reply:
Looks like `state` could have multiple properties set and this branch ensures we only keep `alertStates`, maybe to reduce payload sent to clients. Do you know if the state object can become big enough to be problematic ?

Reply:
Ahh ok, that would make sense, thanks! I will keep this code around, at a first glance I wasn't sure what it was doing but it's clear now. 

>Do you know if the state object can become big enough to be problematic ?

It's possible for sure, there is a circuit breaker at the framework level to prevent more than 1,000 alert objects (and state objects) by default, so it is capped at some point but we haven't seen many issues around the storage size yet.

Reply:
@klacabane I added back the code here: https://github.com/elastic/kibana/pull/162425/commits/c05457212ae707326a90321b7da382017bdcd096. I did it in a way that doesn't mutate the `accum[instanceId]` object to keep TypeScript happy (since `RawAlertInstance` is now a `ReadOnly` type).

Reply:
Looks great thanks!